### ISSUE TITLE
refactor(grid): give the grid layering a contract or collapse it; cell-centred `lat_T`; remove the stub modules and phantom `CylindricalGrid`/`SphericalMask2D` references; fix `spherical_area_weights` pole guard and float32 metadata

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#322]
Part of #280.

## Layering [medium]

`ArakawaCGrid*` is used only as a type hint (`reductions.py`, `advection/reconstruction.py`); `CurvilinearGrid*` only in `operators/interpolation.py`. `area_weights` / `volume_weights` accept `ArakawaCGrid2D` and then `isinstance`-dispatch on the two concrete classes, so the abstract layer adds no polymorphism. `CurvilinearGrid` holds `Lx, dx` but no behaviour, and promises a uniform `dx` that a lat-lon grid does not have (`SphericalGrid.dx = R·dlon` is the equatorial value; `dx_T`, `min_cell_width`, `max_aspect` exist because `dx` lies). Give the layer a contract (`cell_widths()`, `area_weights()`, `face_lengths()` methods dispatched by class) or collapse it to the two concrete families.

`grid/cylindrical.py`, `mask/spherical.py`, `mask/cylindrical.py` are docstring-only modules with no classes; the CHANGELOG names a `CylindricalGrid` stub class that does not exist; eleven operator docstrings promise a `SphericalMask2D` "follow-up" with no tracking issue. Remove the stubs, fix the CHANGELOG, and either open one tracking issue for `SphericalMask2D` or drop the promise.

`CartesianGrid2D(Nx=10, Ny=10, Lx=1, Ly=1, dx=0.5, dy=0.5)` is accepted (`dx·(Nx−2) = 4 ≠ Lx`); no grid has a `__check_init__`. Cartesian docstrings place the first interior T point at `i·dx`; spherical at `lat_min`; a shared cell-centred rule `(i − ½)·dx` would settle both.

## Spherical latitude offset [medium, bug]

`grid/spherical.py:135`: `lat_1d = lat_min + (arange(Ny) − 1)·dlat` is the periodic convention applied to a bounded axis. T rows run from `lat_min` to `lat_max − dlat`:

```
from_interior(48, 24, (0,360), (-80,80)):  lat_T interior rows = -80.0 … 73.33 ; ghost S = -86.67, ghost N = 80.0
from_interior(8, 6, (0,360), (-90,90)):    lat_T rows = [-120, -90, -60, -30, 0, 30, 60, 90]
   cos_lat_T[j=1] (on the pole) = -4.4e-08 ; ghost j=0 cos = -0.5, area weight = -0.2056
   sum of interior area weights = 12.278 vs 4π = 12.566 (-2.3%) ; min_cell_width = 0.0, max_aspect = inf
```

`tests/test_spherical_cell_widths.py:21` says the fixture is "symmetric about the equator with an equator row"; it is not. Most of the pole-guard machinery (`_degenerate_width`, `_safe_div_cos`) exists to cope with this. Fix: `lat_T[j] = lat_min + (j − ½)·dlat` (symmetric rows, no row on a pole, ghosts at most `dlat/2` past it). User-visible; CHANGELOG.

## Small [low]

- `spherical_area_weights` pole guard `|cos| < 1e-12` (`reductions.py:84`) does not fire in float32 (`cos(±π/2) = −4.37e-8`) and returns negative ghost-row weights; reuse `_degenerate_width`'s `8·eps·nominal` rule and clamp to 0. Midpoint area `R² cos_T dlon dlat` overestimates the exact `R² dlon (sin lat_N − sin lat_S)` by `dlat/(2 sin(dlat/2))` = 1.00056 at 6.67°; the exact form is free and makes `area_sum` exact for constants.
- `spherical.py:115-129`: `jnp.deg2rad` on Python floats is float32; `dlon`, `dlat`, `Lx`, `Ly` inherit 2.8e-8 relative error (Earth `Lx` off by 1.11 m). Use `math.radians`.
- `type Nirr = int` (`mask/cartesian.py:49`) breaks `typing.get_type_hints(Mask2D)`; use the string `"Nirr"`.
- `Mask3D` irrbound uses a 3×3×3 neighbourhood while corners are per-level (`mask/cartesian.py:1219`); decide and document.
- Ghost ring is never enforced or checked as land (`from_dimensions` leaves it wet; `count_contiguous` counts ghost cells so a half-width-2 stencil at `i=1` would read `i=−1`); add a `ghost_is_land` check or document.

## Acceptance criteria

- [ ] `lat_T` symmetric about the equator for a symmetric `lat_range`; no T row on a pole; `sum(area_weights) == 4πR²` to 1e-12 with the exact sine form.
- [ ] Inconsistent metric raises in `__check_init__`.
- [ ] No stub modules; no docstring references a class that does not exist.
